arwen/cryptoapi/cryptoei.go

```diff
@@ -287,3 +348,342 @@ func verifySecp256k1(
 
 	return 0
 }
+
+//export addEC
+func addEC(
+	context unsafe.Pointer,
+	xResultHandle int32,
+	yResultHandle int32,
+	fieldOrder int32,
+	basePointOrder int32,
+	eqConstant int32,
+	xBasePoint int32,
+	yBasePoint int32,
+	sizeOfField int32,
+	fstPointX int32,
+	fstPointY int32,
+	sndPointX int32,
+	sndPointY int32,
+) {
+	bigInt := arwen.GetBigIntContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().CryptoAPICost.SHA256
+	metering.UseGas(gasToUse)
+
+	xResult, yResult, P := bigInt.GetThree(xResultHandle, yResultHandle, fieldOrder)
+	N, B, Gx := bigInt.GetThree(basePointOrder, eqConstant, xBasePoint)
+	Gy, x1, y1 := bigInt.GetThree(yBasePoint, fstPointX, fstPointY)
+	x2, y2 := bigInt.GetTwo(sndPointX, sndPointY)
+	bigInt.ConsumeGasForBigIntCopy(xResult, yResult, P, N, B, Gx, Gy, x1, y1, x2, y2)
+
+	xResult, yResult = elliptic_curve.Add(P, N, B, Gx, Gy, int(sizeOfField), x1, y1, x2, y2)
+}
+
+//export doubleEC
+func doubleEC(
+	context unsafe.Pointer,
+	xResultHandle int32,
+	yResultHandle int32,
+	fieldOrder int32,
+	basePointOrder int32,
+	eqConstant int32,
+	xBasePoint int32,
+	yBasePoint int32,
+	sizeOfField int32,
+	pointX int32,
+	pointY int32,
+) {
+	bigInt := arwen.GetBigIntContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().CryptoAPICost.SHA256
+	metering.UseGas(gasToUse)
+
+	xResult, yResult, P := bigInt.GetThree(xResultHandle, yResultHandle, fieldOrder)
+	N, B, Gx := bigInt.GetThree(basePointOrder, eqConstant, xBasePoint)
+	Gy, x, y := bigInt.GetThree(yBasePoint, pointX, pointY)
+	bigInt.ConsumeGasForBigIntCopy(xResult, yResult, P, N, B, Gx, Gy, x, y)
+
+	xResult, yResult = elliptic_curve.Double(P, N, B, Gx, Gy, int(sizeOfField), x, y)
+}
+
+//export isOnCurveEC
+func isOnCurveEC(
+	context unsafe.Pointer,
```

Comment:
please make a difference all over the file between input with big Int handle and not handles. It is hard to read into.